### PR TITLE
JAMES-2434 IsFromMailingList matcher

### PR DESCRIPTION
**Pull request by Maxxx873**
* Implemented IsFromMailingList matcher with using AutomaticallySentMailDetector::isMailingList utility. This matcher returns recipients of the mail if the mail is from a mailing list.
* Fixed IsSingleRecipient matcher - for methods that return an array, prefer returning an empty array over a null value.

### CODE REVIEW

File: mailet/standard/src/main/java/org/apache/james/transport/matchers/IsFromMailingList.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
+package org.apache.james.transport.matchers;
```
Comment (chibenwa) [inline mailet/standard/src/main/java/org/apache/james/transport/matchers/IsFromMailingList.java:1]: You are missing the compulsary Apache V2 license header comment in this file.

File: mailet/standard/src/test/java/org/apache/james/transport/matchers/IsFromMailingListTest.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
+package org.apache.james.transport.matchers;
```
Comment (chibenwa) [inline mailet/standard/src/test/java/org/apache/james/transport/matchers/IsFromMailingListTest.java:1]: You are missing the compulsary Apache V2 license header comment in this file.

**Review by chibenwa — APPROVED** [review]
@ licenses to add but overall thanks for this great contribution!

**Comment by Maxxx873** [comment]
> 2 licenses to add but overall thanks for this great contribution!

Thank you for taking the time to review my code. I appreciate your feedback and will use it to improve my work in the future.

**Review by chibenwa — APPROVED** [review]

**Review by Arsnael — APPROVED** [review]
Thank you for the nice contribution @Maxxx873 , the code looks good to me :)

**Comment by Arsnael** [comment]
```
20:46:20,494 [ERROR] Failed to execute goal org.apache.maven.plugins:maven-checkstyle-plugin:3.1.2:check (check-style) on project apache-mailet-standard: You have 2 Checkstyle violations. -> [Help 1]
```

There is some checkstyle issue with your code (we have a maven plugin doing some strict code style checking following this file: https://github.com/apache/james-project/blob/master/checkstyle.xml during the build)

I would guess likely some unused imports in the files in your PR that you added/modified. Please help fixing this :)

**Comment by Maxxx873** [comment]
> ```
> 20:46:20,494 [ERROR] Failed to execute goal org.apache.maven.plugins:maven-checkstyle-plugin:3.1.2:check (check-style) on project apache-mailet-standard: You have 2 Checkstyle violations. -> [Help 1]
> ```
> 
> There is some checkstyle issue with your code (we have a maven plugin doing some strict code style checking following this file: https://github.com/apache/james-project/blob/master/checkstyle.xml during the build)
> 
> I would guess likely some unused imports in the files in your PR that you added/modified. Please help fixing this :)

Sorry, I'm just used to a different import order :)

**Comment by Arsnael** [comment]
> Sorry, I'm just used to a different import order :)

No worries :)

**Pull request merged**